src/stacked_borrows.rs

```diff
@@ -702,6 +741,18 @@ trait EvalContextPrivExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             );
             return Ok(());
         }
+        if matches!(new_tag, SbTag::Wildcard(_)) {
+            // Nothing to do for wildcard tags.
+            trace!(
+                "reborrow of wildcard: {} reference {:?} derived from {:?} (pointee {})",
+                kind,
+                new_tag,
+                place.ptr,
+                place.layout.ty,
+            );
+            return Ok(());
+        }
+
         let (alloc_id, base_offset, ptr) = this.ptr_get_alloc_id(place.ptr)?;
```

Comment:
There's a problem further down this function: it calls `visit_freeze_sensitive`, which expects the `place.ptr` to have an `AllocId`. Previously that was guaranteed if the `ptr_get_alloc_id` worked here, but now it might be that `place.ptr` is a wildcard and it computes the AllocId on the fly here. Then later the `into_pointer_or_addr().unwrap()` in `visit_freeze_sensitive` will fail.

We probably should adjust `visit_freeze_sensitive` such that it doesn't do that any more, and instead just works with offsets all the time. The only thing it really needs from the pointer is its absolute address, and that is still easy to get from a `Pointer<Option<Tag>>`... we just don't have a convenient method for it.

Reply:
https://github.com/rust-lang/miri/pull/2071 should now take care of that.